src/zlib/reader.cr

```diff
@@ -79,18 +81,30 @@ class Zlib::Reader < IO
   end
 
   # Always raises `IO::Error` because this is a read-only `IO`.
-  def write(slice : Bytes)
+  def unbuffered_write(slice : Bytes)
     raise IO::Error.new "Can't write to Zlib::Reader"
   end
 
-  def close
+  def unbuffered_flush
+    raise IO::Error.new "Can't flush Zlib::Reader"
+  end
+
+  def unbuffered_close
     return if @closed
     @closed = true
 
     @flate_io.close
     @io.close if @sync_close
   end
 
+  def unbuffered_rewind
+    raise IO::Error.new "Closed stream" if closed?
```

Comment:
ditto

Reply:
Yeah, it was like that but I had to rollback to a previous version and I forgot to add that.

Reply:
Someone can send a PR, or commit directly to master with it (I'm trying but I can't).

Reply:
#6958 :)